TST fix tests for JAX-Galsim

beckermr opened the pull request:
This PR has two main features:

1. It has a generalization of the concept of a "galsim backend" with associated functions for the test suite only. In the test suite only, the backend is used to adjust the tests as needed for the jax-based galsim implementation.

2. It has adjustments of the tests for jax-galsim. Most of them are related to cases where jax cannot raise the same errors (e.g., for checking argument types) or when the fact that jax arrays cannot be changed in place causes the APIs to differ.

beckermr commented:
The commits here need to be rebased onto main I think.

beckermr commented on tests/galsim_test_helpers.py:
```diff
@@ -30,6 +30,30 @@
+    "drawNoise",
+    "check_pickle",
+    "check_all_diff",
+    "funcname",
```
```suggestion
```

beckermr commented on tests/galsim_test_helpers.py:
```diff
@@ -30,6 +30,29 @@
+    "CaptureLog",
+    "assert_raises",
+    "assert_warns",
+    "profile",
```
```suggestion
```

beckermr committed:
array tests

beckermr committed:
jax no warn

beckermr committed:
jax no warn

beckermr committed:
jax no warn

beckermr committed:
jax no warn

beckermr committed:
jax no warn

beckermr committed:
jax no fwd fft complex

beckermr committed:
jax no fwd fft complex

beckermr committed:
jax no fwd fft complex

beckermr committed:
array compare

beckermr committed:
array compare

beckermr committed:
jax rounds a bit different

beckermr committed:
jax rounds a bit different

beckermr committed:
jax rounds a bit different

beckermr committed:
make_const not supported

beckermr committed:
paths

beckermr committed:
paths

beckermr committed:
paths

beckermr committed:
paths

beckermr committed:
paths

beckermr committed:
paths

beckermr committed:
paths

beckermr committed:
support make const

beckermr committed:
support make const

beckermr committed:
support make const

beckermr committed:
support make const

beckermr committed:
support make const

beckermr committed:
support make const

beckermr committed:
no references

beckermr committed:
no references

beckermr committed:
no references

beckermr committed:
no references

beckermr committed:
no references

beckermr committed:
no references

beckermr committed:
no references

beckermr committed:
no references

beckermr committed:
no references

beckermr committed:
no references

beckermr committed:
no references

beckermr committed:
weird failure

beckermr committed:
weird failure

beckermr committed:
weird failure

beckermr committed:
weird failure

beckermr committed:
weird failure

beckermr committed:
weird failure

beckermr committed:
weird failure

beckermr committed:
weird failure

beckermr committed:
weird failure

beckermr committed:
weird failure

beckermr committed:
paths

beckermr committed:
paths

beckermr committed:
paths

beckermr committed:
paths

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
paths

beckermr committed:
paths

beckermr committed:
paths

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
norefs

beckermr committed:
no raise

beckermr committed:
no raise

beckermr committed:
no raise

beckermr committed:
no raise

beckermr committed:
no raise

beckermr committed:
no raise

beckermr committed:
Merge branch 'main' into fix_tests

beckermr committed:
no raise

beckermr committed:
paths

beckermr committed:
Merge branch 'fix_tests' of https://github.com/GalSim-developers/GalSim into fix_tests

beckermr committed:
paths

beckermr committed:
no refs

beckermr committed:
no refs

beckermr committed:
array asserts

beckermr commented on tests/test_random.py:
```diff
@@ -48,42 +48,80 @@
-chi2N = 30
-# Tabulated results for Chi2
-chi2Result = (32.209933900954049, 50.040002656028513, 24.301442486313896)
+if not hasattr(galsim, "_galsim"):
```
```suggestion
if hasattr(galsim, "_galsim"):
```

beckermr committed:
Update test_random.py

beckermr committed:
no views

beckermr committed:
Merge branch 'fix_tests' of https://github.com/GalSim-developers/GalSim into fix_tests

beckermr committed:
no views

beckermr committed:
no views

beckermr committed:
no views

beckermr committed:
no views

beckermr committed:
no views

beckermr committed:
no views

beckermr committed:
no views

beckermr committed:
no views

beckermr committed:
no views

beckermr committed:
no views

beckermr committed:
no views

beckermr committed:
no views

beckermr committed:
no views

beckermr committed:
no views

beckermr committed:
no views

beckermr committed:
no index errors

beckermr committed:
bool env

beckermr committed:
bool env

beckermr committed:
zernike import

beckermr committed:
bug in test of randoms

beckermr committed:
remove deprecation warnings

beckermr commented on tests/test_deprecated.py:
```diff
@@ -25,8 +25,7 @@
 def check_dep(f, *args, **kwargs):
-    """Check that some function raises a GalSimDeprecationWarning as a warning, but not an error.
-    """
+    """Check that some function raises a GalSimDeprecationWarning as a warning, but not an error."""
```
```suggestion
    """Check that some function raises a GalSimDeprecationWarning as a warning, but not an error.
"""
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -37,70 +36,89 @@ def check_dep(f, *args, **kwargs):
     check_dep(galsim.GSParams, allowed_flux_variation=0.90)
     check_dep(galsim.GSParams, range_division_for_extrema=50)
-    check_dep(galsim.GSParams, small_fraction_of_flux=1.e-6)
+    check_dep(galsim.GSParams, small_fraction_of_flux=1.0e-6)
```
```suggestion
    check_dep(galsim.GSParams, small_fraction_of_flux=1.e-6)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -37,70 +36,89 @@ def check_dep(f, *args, **kwargs):
 @timer
 def test_phase_psf():
-    atm = galsim.Atmosphere(screen_size=10.0, altitude=0, r0_500=0.15, suppress_warning=True)
+    atm = galsim.Atmosphere(
+        screen_size=10.0, altitude=0, r0_500=0.15, suppress_warning=True
+    )
```
```suggestion
    atm = galsim.Atmosphere(screen_size=10.0, altitude=0, r0_500=0.15, suppress_warning=True)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -37,70 +36,89 @@ def check_dep(f, *args, **kwargs):
     check_dep(galsim.PhaseScreenPSF.__getattribute__, psf, "img")
     check_dep(galsim.PhaseScreenPSF.__getattribute__, psf, "finalized")
 
+
```
```suggestion
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -37,70 +36,89 @@ def check_dep(f, *args, **kwargs):
 def test_noise():
-    real_gal_dir = os.path.join('..','examples','data')
-    real_gal_cat = 'real_galaxy_catalog_23.5_example.fits'
+    real_gal_dir = os.path.join("..", "examples", "data")
+    real_gal_cat = "real_galaxy_catalog_23.5_example.fits"
```
```suggestion
    real_gal_dir = os.path.join('..','examples','data')
    real_gal_cat = 'real_galaxy_catalog_23.5_example.fits'
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -37,70 +36,89 @@ def check_dep(f, *args, **kwargs):
+    real_gal_cat = "real_galaxy_catalog_23.5_example.fits"
     real_cat = galsim.RealGalaxyCatalog(
-        dir=real_gal_dir, file_name=real_gal_cat, preload=True)
+        dir=real_gal_dir, file_name=real_gal_cat, preload=True
+    )
```
```suggestion
        dir=real_gal_dir, file_name=real_gal_cat, preload=True)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -37,70 +36,89 @@ def check_dep(f, *args, **kwargs):
+    )
 
-    test_seed=987654
+    test_seed = 987654
```
```suggestion
    test_seed=987654
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -37,70 +36,89 @@ def check_dep(f, *args, **kwargs):
     # Check the variance:
     var_1 = cf_1.getVariance()
-    assert var_1==var_2,'Inconsistent noise variance from getNoise and getNoiseProperties'
+    assert (
+        var_1 == var_2
+    ), "Inconsistent noise variance from getNoise and getNoiseProperties"
```
```suggestion
    assert var_1==var_2,'Inconsistent noise variance from getNoise and getNoiseProperties'
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -110,26 +128,33 @@ def test_randwalk_defaults():
 
     # try constructing with mostly defaults
-    npoints=100
+    npoints = 100
```
```suggestion
    npoints=100
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -110,26 +128,33 @@ def test_randwalk_defaults():
     hlr = 8.0
     rng = galsim.BaseDeviate(1234)
-    rw=check_dep(galsim.RandomWalk, npoints, half_light_radius=hlr, rng=rng)
+    rw = check_dep(galsim.RandomWalk, npoints, half_light_radius=hlr, rng=rng)
```
```suggestion
    rw=check_dep(galsim.RandomWalk, npoints, half_light_radius=hlr, rng=rng)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -110,26 +128,33 @@ def test_randwalk_defaults():
 
-    nobj=len(rw.points)
-    assert nobj == npoints,"expected %d objects, got %d" % (npoints, nobj)
+    nobj = len(rw.points)
+    assert nobj == npoints, "expected %d objects, got %d" % (npoints, nobj)
```
```suggestion
    nobj=len(rw.points)
    assert nobj == npoints,"expected %d objects, got %d" % (npoints, nobj)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -110,26 +128,33 @@ def test_randwalk_defaults():
+    np.testing.assert_almost_equal(rw.centroid.y, np.mean(pts[:, 1]))
 
-    gsp = galsim.GSParams(xvalue_accuracy=1.e-8, kvalue_accuracy=1.e-8)
+    gsp = galsim.GSParams(xvalue_accuracy=1.0e-8, kvalue_accuracy=1.0e-8)
```
```suggestion
    gsp = galsim.GSParams(xvalue_accuracy=1.e-8, kvalue_accuracy=1.e-8)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -143,7 +168,7 @@ def test_randwalk_defaults():
     # Check that image is not sensitive to use of rng by other objects.
     rng3 = galsim.BaseDeviate(1234)
-    rw3=check_dep(galsim.RandomWalk, npoints, half_light_radius=hlr, rng=rng3)
+    rw3 = check_dep(galsim.RandomWalk, npoints, half_light_radius=hlr, rng=rng3)
```
```suggestion
    rw3=check_dep(galsim.RandomWalk, npoints, half_light_radius=hlr, rng=rng3)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -153,7 +178,7 @@ def test_randwalk_defaults():
     # Run some basic tests of correctness
     check_basic(conv1, "RandomWalk")
-    im = galsim.ImageD(64,64, scale=0.5)
+    im = galsim.ImageD(64, 64, scale=0.5)
```
```suggestion
    im = galsim.ImageD(64,64, scale=0.5)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -168,38 +193,45 @@ def test_randwalk_repr():
     """
 
-    npoints=100
+    npoints = 100
```
```suggestion
    npoints=100
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -168,38 +193,45 @@ def test_randwalk_repr():
         flux=flux,
     )
-    rw2=check_dep(galsim.RandomWalk,
+    rw2 = check_dep(
+        galsim.RandomWalk,
```
```suggestion
    rw2=check_dep(galsim.RandomWalk,
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -168,38 +193,45 @@ def test_randwalk_repr():
     for rw in (rw1, rw2):
-
-
         # just make sure str() works, don't require eval to give
```
```suggestion
        
        
        # just make sure str() works, don't require eval to give
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -168,38 +193,45 @@ def test_randwalk_repr():
         # just make sure str() works, don't require eval to give
         # a consistent object back
-        st=str(rw)
+        st = str(rw)
```
```suggestion
        st=str(rw)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -208,136 +240,177 @@ def test_randwalk_config():
 
-    hlr=2.0
-    flux=np.pi
+    hlr = 2.0
+    flux = np.pi
```
```suggestion
    hlr=2.0
    flux=np.pi
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -208,136 +240,177 @@ def test_randwalk_config():
-        'npoints':100,
-        'half_light_radius':hlr,
-        'flux':flux,
+        "type": "RandomWalk",
+        "npoints": 100,
+        "half_light_radius": hlr,
+        "flux": flux,
```
```suggestion
        'type': 'RandomWalk',
        'npoints': 100,
        'half_light_radius': hlr,
        'flux': flux,
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -208,136 +240,177 @@ def test_randwalk_config():
-            'half_light_radius': hlr,
-            'flux': flux,
-        }
+        "type": "RandomWalk",
+        "npoints": 150,
+        "profile": {
+            "type": "Exponential",
+            "half_light_radius": hlr,
+            "flux": flux,
+        },
```
```suggestion
        'type': 'RandomWalk',
        'npoints': 150,
        'profile': {
            'type': 'Exponential',
            'half_light_radius': hlr,
            'flux': flux,
        }
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -208,136 +240,177 @@ def test_randwalk_config():
-        config={
-            'gal':gal_config,
-            'rng':galsim.BaseDeviate(31415),
+        config = {
+            "gal": gal_config,
+            "rng": galsim.BaseDeviate(31415),
```
```suggestion
        config={
            'gal': gal_config,
            'rng': galsim.BaseDeviate(31415),
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -208,136 +240,177 @@ def test_randwalk_config():
         }
 
-        rwc = check_dep(galsim.config.BuildGSObject, config, 'gal')[0]
+        rwc = check_dep(galsim.config.BuildGSObject, config, "gal")[0]
```
```suggestion
        rwc = check_dep(galsim.config.BuildGSObject, config, 'gal')[0]
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -208,136 +240,177 @@ def test_randwalk_config():
 
-        rw = check_dep(galsim.RandomWalk,
-            gal_config['npoints'],
+        rw = check_dep(
+            galsim.RandomWalk,
+            gal_config["npoints"],
```
```suggestion
        rw = check_dep(galsim.RandomWalk,
            gal_config['npoints'],
```

beckermr reviewed (commented):
Putting extraneous changes back. IDK how these got in here.

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -789,6 +930,6 @@ def test_photon_array_correlated():
 
 if __name__ == "__main__":
-    testfns = [v for k, v in vars().items() if k[:5] == 'test_' and callable(v)]
+    testfns = [v for k, v in vars().items() if k[:5] == "test_" and callable(v)]
```
```suggestion
    testfns = [v for k, v in vars().items() if k[:5] == 'test_' and callable(v)]
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -639,65 +771,74 @@ def test_photon_array_depr():
 
     # And check deprecated SED attribute.
-    sed = check_dep(getattr, PSF, 'SED')
+    sed = check_dep(getattr, PSF, "SED")
```
```suggestion
    sed = check_dep(getattr, PSF, 'SED')
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -639,65 +771,74 @@ def test_photon_array_depr():
+    sed = check_dep(getattr, PSF, "SED")
     assert sed == PSF.sed
-    sed1 = check_dep(getattr, PSF1, 'SED')
+    sed1 = check_dep(getattr, PSF1, "SED")
```
```suggestion
    sed1 = check_dep(getattr, PSF1, 'SED')
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -639,65 +771,74 @@ def test_photon_array_depr():
     assert sed1 == PSF1.sed
-    sed2 = check_dep(getattr, mono_PSF, 'SED')
-    assert sed1 == mono_PSF.sed == galsim.SED(1, 'nm', '1')
+    sed2 = check_dep(getattr, mono_PSF, "SED")
+    assert sed1 == mono_PSF.sed == galsim.SED(1, "nm", "1")
+
```
```suggestion
    sed2 = check_dep(getattr, mono_PSF, 'SED')
    assert sed1 == mono_PSF.sed == galsim.SED(1, 'nm', '1')
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -639,65 +771,74 @@ def test_photon_array_depr():
     config = {
-        'modules' : ['galsim.roman'],
-        'psf' : { 'type' : 'RomanPSF', 'SCA': 4, 'bandpass': 'W149' }
+        "modules": ["galsim.roman"],
+        "psf": {"type": "RomanPSF", "SCA": 4, "bandpass": "W149"},
```
```suggestion
        'modules': ['galsim.roman'],
        'psf': {'type': 'RomanPSF', 'SCA': 4, 'bandpass': 'W149'},
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -639,65 +771,74 @@ def test_photon_array_depr():
-    psf2 = check_dep(galsim.roman.getPSF, SCA=4, bandpass='W149')
-    print('psf1 = ',str(psf1))
-    print('psf2 = ',str(psf2))
+    psf1 = check_dep(galsim.config.BuildGSObject, config, "psf")[0]
+    psf2 = check_dep(galsim.roman.getPSF, SCA=4, bandpass="W149")
+    print("psf1 = ", str(psf1))
+    print("psf2 = ", str(psf2))
```
```suggestion
    psf1 = check_dep(galsim.config.BuildGSObject, config, 'psf')[0]
    psf2 = check_dep(galsim.roman.getPSF, SCA=4, bandpass='W149')
    print('psf1 = ', str(psf1))
    print('psf2 = ', str(psf2))
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -675,29 +815,29 @@ def test_W149():
     config = {
-        'modules' : ['galsim.roman'],
-        'psf' : { 'type' : 'RomanPSF', 'SCA': 4, 'bandpass': 'W149' }
+        'modules': ['galsim.roman'],
+        'psf': {'type': 'RomanPSF', 'SCA': 4, 'bandpass': 'W149'},
```
```suggestion
        'modules' : ['galsim.roman'],
        'psf' : { 'type' : 'RomanPSF', 'SCA': 4, 'bandpass': 'W149' },
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -639,27 +771,35 @@ def test_photon_array_depr():
-    PSF3 = check_dep(galsim.InterpolatedChromaticObject, PSF,
-                     waves=np.linspace(bandpass.blue_limit, bandpass.red_limit, 30),
-                     use_exact_SED=False)
+    PSF3 = check_dep(
+        galsim.InterpolatedChromaticObject,
+        PSF,
+        waves=np.linspace(bandpass.blue_limit, bandpass.red_limit, 30),
+        use_exact_SED=False,
+    )
```
```suggestion
    PSF3 = check_dep(galsim.InterpolatedChromaticObject, PSF,
                     waves=np.linspace(bandpass.blue_limit, bandpass.red_limit, 30),
                     use_exact_SED=False)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -639,27 +771,35 @@ def test_photon_array_depr():
-    PSF2 = check_dep(PSF.interpolate,
-                     waves=np.linspace(bandpass.blue_limit, bandpass.red_limit, 30),
-                     use_exact_SED=False)
+    PSF2 = check_dep(
+        PSF.interpolate,
+        waves=np.linspace(bandpass.blue_limit, bandpass.red_limit, 30),
+        use_exact_SED=False,
+    )
```
```suggestion
    PSF2 = check_dep(PSF.interpolate,
                     waves=np.linspace(bandpass.blue_limit, bandpass.red_limit, 30),
                     use_exact_SED=False)
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -676,7 +811,7 @@ def test_W149():
     config = {
         'modules' : ['galsim.roman'],
-        'psf' : { 'type' : 'RomanPSF', 'SCA': 4, 'bandpass': 'W149' }
+        'psf' : { 'type' : 'RomanPSF', 'SCA': 4, 'bandpass': 'W149' },
```
```suggestion
        'psf' : { 'type' : 'RomanPSF', 'SCA': 4, 'bandpass': 'W149' }
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -26,7 +26,7 @@
 def check_dep(f, *args, **kwargs):
     """Check that some function raises a GalSimDeprecationWarning as a warning, but not an error.
-    """
+"""
```
```suggestion
    """
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -95,12 +96,24 @@ def test_noise():
+    )
     cf_2 = cf_2.withVariance(var_2)
-    assert cf_1==cf_2,'Inconsistent noise properties from getNoise and getNoiseProperties'
+    assert (
+        cf_1 == cf_2
+    ), "Inconsistent noise properties from getNoise and getNoiseProperties"
+
```
```suggestion
    assert cf_1==cf_2, 'Inconsistent noise properties from getNoise and getNoiseProperties'
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -95,12 +96,24 @@ def test_noise():
-                                  calculate_maxk=False, x_interpolant='linear')
-    cf_2 = check_dep(galsim.correlatednoise._BaseCorrelatedNoise,
-                     galsim.BaseDeviate(test_seed), ii, im_2.wcs)
+    ii = galsim.InterpolatedImage(
+        im_2,
+        normalization="sb",
+        calculate_stepk=False,
+        calculate_maxk=False,
+        x_interpolant="linear",
+    )
+    cf_2 = check_dep(
+        galsim.correlatednoise._BaseCorrelatedNoise,
+        galsim.BaseDeviate(test_seed),
+        ii,
+        im_2.wcs,
+    )
```
```suggestion
    ii = galsim.InterpolatedImage(im_2, normalization='sb', calculate_stepk=False,
        calculate_maxk=False, x_interpolant='linear')
    cf_2 = check_dep(galsim.correlatednoise._BaseCorrelatedNoise,
                     galsim.BaseDeviate(test_seed), ii, im_2.wcs)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -115,21 +128,28 @@ def test_randwalk_defaults():
-    assert rw.npoints==npoints,"expected npoints==%d, got %d" % (npoints, rw.npoints)
-    assert rw.input_half_light_radius==hlr,\
-        "expected hlr==%g, got %g" % (hlr, rw.input_half_light_radius)
+    assert rw.npoints == npoints, "expected npoints==%d, got %d" % (npoints, rw.npoints)
```
```suggestion
    assert rw.npoints==npoints,"expected npoints==%d, got %d" % (npoints, rw.npoints)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -115,21 +128,28 @@ def test_randwalk_defaults():
-    assert rw.input_half_light_radius==hlr,\
-        "expected hlr==%g, got %g" % (hlr, rw.input_half_light_radius)
+    assert rw.npoints == npoints, "expected npoints==%d, got %d" % (npoints, rw.npoints)
+    assert rw.input_half_light_radius == hlr, "expected hlr==%g, got %g" % (
+        hlr,
+        rw.input_half_light_radius,
+    )
```
```suggestion
    assert rw.input_half_light_radius==hlr,\
        "expected hlr==%g, got %g" % (hlr, rw.input_half_light_radius)
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -96,11 +97,11 @@ def test_noise():
     # Check the image:
     ii = galsim.InterpolatedImage(im_2, normalization='sb', calculate_stepk=False,
-                                  calculate_maxk=False, x_interpolant='linear')
+        calculate_maxk=False, x_interpolant='linear')
```
```suggestion
                                  calculate_maxk=False, x_interpolant='linear')
```

beckermr committed:
Update tests/test_deprecated.py

beckermr commented on tests/test_deprecated.py:
```diff
@@ -100,7 +101,7 @@ def test_noise():
                      galsim.BaseDeviate(test_seed), ii, im_2.wcs)
     cf_2 = cf_2.withVariance(var_2)
-    assert cf_1==cf_2,'Inconsistent noise properties from getNoise and getNoiseProperties'
+    assert cf_1==cf_2, 'Inconsistent noise properties from getNoise and getNoiseProperties'
```
```suggestion
    assert cf_1==cf_2,'Inconsistent noise properties from getNoise and getNoiseProperties'
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -122,14 +123,19 @@ def test_randwalk_defaults():
-    assert pts.shape == (npoints,2),"expected (%d,2) shape for points, got %s" % (npoints, pts.shape)
-    np.testing.assert_almost_equal(rw.centroid.x, np.mean(pts[:,0]))
-    np.testing.assert_almost_equal(rw.centroid.y, np.mean(pts[:,1]))
+    pts = rw.points
+    assert pts.shape == (npoints, 2), "expected (%d,2) shape for points, got %s" % (
+        npoints,
+        pts.shape,
+    )
```
```suggestion
    pts=rw.points
    assert pts.shape == (npoints, 2),"expected (%d,2) shape for points, got %s" % (npoints, pts.shape)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -122,14 +123,19 @@ def test_randwalk_defaults():
     gsp = galsim.GSParams(xvalue_accuracy=1.e-8, kvalue_accuracy=1.e-8)
     rng2 = galsim.BaseDeviate(1234)
-    rw2 = check_dep(galsim.RandomWalk, npoints, half_light_radius=hlr, rng=rng2, gsparams=gsp)
+    rw2 = check_dep(
+        galsim.RandomWalk, npoints, half_light_radius=hlr, rng=rng2, gsparams=gsp
+    )
```
```suggestion
    rw2 = check_dep(galsim.RandomWalk, npoints, half_light_radius=hlr, rng=rng2, gsparams=gsp)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -170,8 +176,9 @@ def test_randwalk_repr():
     hlr = 8.0
-    flux=1
-    rw1=check_dep(galsim.RandomWalk,
+    flux = 1
+    rw1 = check_dep(
+        galsim.RandomWalk,
```
```suggestion
    flux=1
    rw1=check_dep(galsim.RandomWalk,
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -182,8 +189,8 @@ def test_randwalk_repr():
     for rw in (rw1, rw2):
-
-
+        
+        
```
```suggestion


```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -211,14 +225,14 @@ def test_randwalk_config():
-        'npoints':100,
-        'half_light_radius':hlr,
-        'flux':flux,
+        'type': 'RandomWalk',
+        'npoints': 100,
+        'half_light_radius': hlr,
+        'flux': flux,
```
```suggestion
        'type':'RandomWalk',
        'npoints':100,
        'half_light_radius':hlr,
        'flux':flux,
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -228,8 +242,8 @@ def test_randwalk_config():
         config={
-            'gal':gal_config,
-            'rng':galsim.BaseDeviate(31415),
+            'gal': gal_config,
+            'rng': galsim.BaseDeviate(31415),
```
```suggestion
            'gal':gal_config,
            'rng':galsim.BaseDeviate(31415),
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -241,103 +255,143 @@ def test_randwalk_config():
     # Now some CelestialWCS types
-    cubic_u = Cubic(2.9e-5, 2000., 'u')
-    cubic_v = Cubic(-3.7e-5, 2000., 'v')
+    cubic_u = Cubic(2.9e-5, 2000.0, 'u')
+    cubic_v = Cubic(-3.7e-5, 2000.0, 'v')
```
```suggestion
    cubic_u = Cubic(2.9e-5, 2000., 'u')
    cubic_v = Cubic(-3.7e-5, 2000., 'v')
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -241,103 +255,143 @@ def test_randwalk_config():
 def test_wfirst():
-    """Test that the deprecated wfirst module works like the new roman module.
-    """
+    """Test that the deprecated wfirst module works like the new roman module."""
```
```suggestion
    """Test that the deprecated wfirst module works like the new roman module.
    """
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -639,16 +752,17 @@ def test_photon_array_depr():
     PSF = PSF * 1.0
     PSF1 = PSF.interpolate(waves=np.linspace(bandpass.blue_limit, bandpass.red_limit, 30),
-                          use_exact_sed=False)
+                           use_exact_sed=False)
```
```suggestion
                          use_exact_sed=False)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -639,16 +752,17 @@ def test_photon_array_depr():
     # This is based on a snippet of test_chromatic_flux in test_chromatic.py.
 
-    bulge_SED = galsim.SED('CWW_E_ext.sed', wave_type='ang', flux_type='flambda')
+    bulge_SED = galsim.SED("CWW_E_ext.sed", wave_type="ang", flux_type="flambda")
```
```suggestion
    bulge_SED = galsim.SED('CWW_E_ext.sed', wave_type='ang', flux_type='flambda')
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -540,96 +619,130 @@ def test_photon_array_depr():
+    assert photon_array.hasAllocatedAngles()
     np.testing.assert_array_equal(photon_array.dxdz, 0.17)
-    np.testing.assert_array_equal(photon_array.dydz, 0.)
+    np.testing.assert_array_equal(photon_array.dydz, 0.0)
```
```suggestion
    np.testing.assert_array_equal(photon_array.dydz, 0.)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -521,7 +600,7 @@ def test_photon_array_depr():
     np.testing.assert_array_equal(photon_array.dydz, 0.59)
 
-    photon_array.wavelength = 500.
+    photon_array.wavelength = 500.0
```
```suggestion
    photon_array.wavelength = 500.
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -512,7 +591,7 @@ def test_photon_array_depr():
     assert len(photon_array.dydz) == nphotons
     np.testing.assert_array_equal(photon_array.dxdz, 0.17)
-    np.testing.assert_array_equal(photon_array.dydz, 0.)
+    np.testing.assert_array_equal(photon_array.dydz, 0.0)
```
```suggestion
    np.testing.assert_array_equal(photon_array.dydz, 0.)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -487,10 +564,12 @@ def test_trapz_basic():
 def test_hsm_depr():
     hsmp = check_dep(galsim.hsm.HSMParams, max_moment_nsig2=25.0)
-    assert hsmp.max_moment_nsig2 == 0.
+    assert hsmp.max_moment_nsig2 == 0.0
+
```
```suggestion
    assert hsmp.max_moment_nsig2 == 0.
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -487,10 +564,12 @@ def test_trapz_basic():
     with assert_raises(TypeError):
         check_dep(galsim.integ.trapz, func, 0.1, 1, points=2.3)
 
+
```
```suggestion
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -444,41 +509,53 @@ def test_surface_ops():
 
     result = check_dep(galsim.integ.trapz, func, 0, 1, np.linspace(0, 1, 100000))
-    expected_val = 1.**3./3.
+    expected_val = 1.0**3.0 / 3.0
```
```suggestion
    expected_val = 1.**3./3.
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -123,9 +124,9 @@ def test_randwalk_defaults():
-    assert pts.shape == (npoints,2),"expected (%d,2) shape for points, got %s" % (npoints, pts.shape)
-    np.testing.assert_almost_equal(rw.centroid.x, np.mean(pts[:,0]))
-    np.testing.assert_almost_equal(rw.centroid.y, np.mean(pts[:,1]))
+    assert pts.shape == (npoints, 2),"expected (%d,2) shape for points, got %s" % (npoints, pts.shape)
+    np.testing.assert_almost_equal(rw.centroid.x, np.mean(pts[:, 0]))
+    np.testing.assert_almost_equal(rw.centroid.y, np.mean(pts[:, 1]))
```
```suggestion
    assert pts.shape == (npoints,2),"expected (%d,2) shape for points, got %s" % (npoints, pts.shape)
    np.testing.assert_almost_equal(rw.centroid.x, np.mean(pts[:,0]))
    np.testing.assert_almost_equal(rw.centroid.y, np.mean(pts[:,1]))
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -192,14 +193,21 @@ def test_randwalk_repr():
 
-        assert new_rw.npoints == rw.npoints,\
-            "expected npoints=%d got %d" % (rw.npoints,new_rw.npoints)
+        assert new_rw.npoints == rw.npoints, "expected npoints=%d got %d" % (
+            rw.npoints,
+            new_rw.npoints,
+        )
```
```suggestion
        assert new_rw.npoints==rw.npoints,\
            "expected npoints=%d got %d" % (rw.npoints,new_rw.npoints)
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -192,14 +193,19 @@ def test_randwalk_repr():
         new_rw = eval(repr(rw))
 
-        assert new_rw.npoints == rw.npoints,\
+        assert new_rw.npoints==rw.npoints,\
```
```suggestion
        assert new_rw.npoints == rw.npoints,\
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -192,14 +193,19 @@ def test_randwalk_repr():
-            mess % (rw.input_half_light_radius,new_rw.input_half_light_radius)
-        assert new_rw.flux == rw.flux,\
-            "expected flux=%.16g got %.16g" % (rw.flux,new_rw.flux)
+        mess = "expected input_half_light_radius=%.16g got %.16g"
+        assert new_rw.input_half_light_radius == rw.input_half_light_radius, mess % (
+            rw.input_half_light_radius,
+            new_rw.input_half_light_radius,
+        )
+        assert new_rw.flux == rw.flux, "expected flux=%.16g got %.16g" % (
+            rw.flux,
+            new_rw.flux,
+        )
+
```
```suggestion
        mess="expected input_half_light_radius=%.16g got %.16g"
        assert new_rw.input_half_light_radius == rw.input_half_light_radius,\
            mess % (rw.input_half_light_radius,new_rw.input_half_light_radius)
        assert new_rw.flux == rw.flux,\
            "expected flux=%.16g got %.16g" % (rw.flux,new_rw.flux)
```

beckermr committed:
Apply suggestions from code review

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -241,34 +242,50 @@ def test_randwalk_config():
         nobj=len(rw.points)
         nobjc=len(rwc.points)
-        assert nobj==nobjc,"expected %d objects, got %d" % (nobj,nobjc)
+        assert nobj == nobjc,"expected %d objects, got %d" % (nobj, nobjc)
```
```suggestion
        assert nobj==nobjc,"expected %d objects, got %d" % (nobj, nobjc)
```

beckermr committed:
Update tests/test_deprecated.py

beckermr commented on tests/test_deprecated.py:
```diff
@@ -241,34 +242,50 @@ def test_randwalk_config():
         nobj=len(rw.points)
         nobjc=len(rwc.points)
-        assert nobj==nobjc,"expected %d objects, got %d" % (nobj,nobjc)
+        assert nobj==nobjc,"expected %d objects, got %d" % (nobj, nobjc)
```
```suggestion
        assert nobj==nobjc,"expected %d objects, got %d" % (nobj,nobjc)
```

beckermr committed:
Update tests/test_deprecated.py

beckermr commented on tests/test_deprecated.py:
```diff
@@ -241,34 +242,50 @@ def test_randwalk_config():
-        ptsc=rwc.points
-        assert (pts.shape == ptsc.shape),\
-                "expected %s shape for points, got %s" % (pts.shape,ptsc.shape)
+        pts = rw.points
+        ptsc = rwc.points
+        assert pts.shape == ptsc.shape, "expected %s shape for points, got %s" % (
+            pts.shape,
+            ptsc.shape,
+        )
+
```
```suggestion
        pts=rw.points
        ptsc=rwc.points
        assert (pts.shape == ptsc.shape),\
                "expected %s shape for points, got %s" % (pts.shape,ptsc.shape)
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -242,10 +243,14 @@ def test_randwalk_config():
 
         assert rw.npoints==rwc.npoints,\
-            "expected npoints==%d, got %d" % (rw.npoints, rwc.npoints)
+            "expected npoints==%d, got %d" % (rw.npoints,rwc.npoints)
```
```suggestion
            "expected npoints==%d, got %d" % (rw.npoints, rwc.npoints)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -283,61 +295,86 @@ def test_withOrigin():
         new_origin = galsim.PositionI(123,321)
         wcs3 = wcs.shiftOrigin(new_origin)
-        assert wcs != wcs3, name+' is not != wcs.shiftOrigin(pos)'
+        assert wcs != wcs3, name + " is not != wcs.shiftOrigin(pos)"
```
```suggestion
        assert wcs != wcs3, name+' is not != wcs.shiftOrigin(pos)'
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -283,61 +295,86 @@ def test_withOrigin():
-        assert wcs != wcs4, name+' is not != wcs.local()'
-        assert wcs4 != wcs, name+' is not != wcs.local() (reverse)'
-        world_pos1 = wcs.toWorld(galsim.PositionD(0,0))
+        assert wcs != wcs4, name + " is not != wcs.local()"
```
```suggestion
        assert wcs != wcs4, name+' is not != wcs.local()'
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -283,61 +295,86 @@ def test_withOrigin():
-        assert wcs4 != wcs, name+' is not != wcs.local() (reverse)'
-        world_pos1 = wcs.toWorld(galsim.PositionD(0,0))
+        assert wcs != wcs4, name + " is not != wcs.local()"
+        assert wcs4 != wcs, name + " is not != wcs.local() (reverse)"
```
```suggestion
        assert wcs4 != wcs, name+' is not != wcs.local() (reverse)'
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -283,61 +295,86 @@ def test_withOrigin():
-        world_pos1 = wcs.toWorld(galsim.PositionD(0,0))
+        assert wcs != wcs4, name + " is not != wcs.local()"
+        assert wcs4 != wcs, name + " is not != wcs.local() (reverse)"
+        world_pos1 = wcs.toWorld(galsim.PositionD(0, 0))
```
```suggestion
        world_pos1 = wcs.toWorld(galsim.PositionD(0,0))
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -283,61 +295,86 @@ def test_withOrigin():
     """
     import galsim.roman
-    check_dep(__import__, 'galsim.wfirst')
+
+    check_dep(__import__, "galsim.wfirst")
```
```suggestion
    check_dep(__import__, 'galsim.wfirst')
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -357,10 +394,13 @@ def test_wfirst():
-                                  galsim.roman.persistence_coefficients)
-    np.testing.assert_array_equal(galsim.wfirst.persistence_fermi_parameters,
-                                  galsim.roman.persistence_fermi_parameters)
+    np.testing.assert_array_equal(
+        galsim.wfirst.persistence_coefficients, galsim.roman.persistence_coefficients
+    )
+    np.testing.assert_array_equal(
+        galsim.wfirst.persistence_fermi_parameters,
+        galsim.roman.persistence_fermi_parameters,
+    )
```
```suggestion
    np.testing.assert_array_equal(galsim.wfirst.persistence_coefficients, 
                                  galsim.roman.persistence_coefficients)
    np.testing.assert_array_equal(galsim.wfirst.persistence_fermi_parameters,
                                  galsim.roman.persistence_fermi_parameters)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -382,42 +422,49 @@ def test_wfirst():
     assert galsim.wfirst.allDetectorEffects is galsim.roman.allDetectorEffects
     assert galsim.wfirst.NLfunc is galsim.roman.NLfunc
 
+
```
```suggestion
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -382,42 +422,49 @@ def test_wfirst():
 def test_roman_psfs():
-    """Test the deprecated high_accuracy and approximate_struts options.
-    """
+    """Test the deprecated high_accuracy and approximate_struts options."""
```
```suggestion
    """Test the deprecated high_accuracy and approximate_struts options.
    """
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -382,42 +422,49 @@ def test_wfirst():
 @timer
 def test_surface_ops():
-
     # Based on test_sensor.py:test_wavelengths_and_angles, but massively simplified.
```
```suggestion

    # Based on test_sensor.py:test_wavelengths_and_angles, but massively simplified.
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -426,14 +473,20 @@ def test_surface_ops():
 
-    sed = galsim.SED('CWW_E_ext.sed', 'nm', 'flambda').thin()
-    bandpass = galsim.Bandpass('LSST_i.dat', 'nm').thin()
+    sed = galsim.SED("CWW_E_ext.sed", "nm", "flambda").thin()
+    bandpass = galsim.Bandpass("LSST_i.dat", "nm").thin()
```
```suggestion
    sed = galsim.SED('CWW_E_ext.sed', 'nm', 'flambda').thin()
    bandpass = galsim.Bandpass('LSST_i.dat', 'nm').thin()
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -426,14 +473,20 @@ def test_surface_ops():
-    im = galsim.Image(63,63, scale=1)
-    check_dep(obj.drawImage, im, method='phot', surface_ops=[sampler, assigner], rng=rng,
-              save_photons=True)
+    im = galsim.Image(63, 63, scale=1)
```
```suggestion
    im = galsim.Image(63,63, scale=1)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -426,14 +473,20 @@ def test_surface_ops():
-    check_dep(obj.drawImage, im, method='phot', surface_ops=[sampler, assigner], rng=rng,
-              save_photons=True)
+    im = galsim.Image(63, 63, scale=1)
+    check_dep(
+        obj.drawImage,
+        im,
+        method="phot",
+        surface_ops=[sampler, assigner],
+        rng=rng,
+        save_photons=True,
+    )
```
```suggestion
    check_dep(obj.drawImage, im, method="phot", surface_ops=[sampler, assigner], rng=rng,
              save_photons=True)
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -444,41 +488,53 @@ def test_surface_ops():
 def test_midpoint_basic():
-    """Test the basic functionality of the midpt() method.
-    """
+    """Test the basic functionality of the midpt() method."""
```
```suggestion
    """Test the basic functionality of the midpt() method.
    """
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -444,41 +488,53 @@ def test_surface_ops():
+    """Test the basic functionality of the midpt() method."""
     # This shouldn't be super accurate, but just make sure it's not really broken.
-    x = 0.01*np.arange(1000)
+    x = 0.01 * np.arange(1000)
```
```suggestion
    x = 0.01*np.arange(1000)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -444,41 +488,53 @@ def test_surface_ops():
     np.testing.assert_almost_equal(
-        result/expected_val, 1.0, decimal=2, verbose=True,
-        err_msg='Simple test of midpt() method failed for f(x)=x^2 from 0 to 10')
+        result / expected_val,
+        1.0,
+        decimal=2,
+        verbose=True,
+        err_msg="Simple test of midpt() method failed for f(x)=x^2 from 0 to 10",
+    )
```
```suggestion
        result/expected_val, 1.0, decimal=2, verbose=True,
        err_msg='Simple test of midpt() method failed for f(x)=x^2 from 0 to 10')
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -444,41 +488,53 @@ def test_surface_ops():
 def test_trapz_basic():
-    """Test the basic functionality of the trapz() method.
-    """
+    """Test the basic functionality of the trapz() method."""
```
```suggestion
    """Test the basic functionality of the trapz() method.
    """
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -444,41 +488,53 @@ def test_surface_ops():
     func = lambda x: x**2
     result = check_dep(galsim.integ.trapz, func, 0, 1)
-    expected_val = 1.**3./3.
+    expected_val = 1.0**3.0 / 3.0
```
```suggestion
    expected_val = 1.**3./3.
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -644,7 +732,7 @@ def test_chromatic_flux():
     zenith_angle = 20 * galsim.degrees
     bandpass = galsim.Bandpass('LSST_i.dat', 'nm').thin()
-    PSF = galsim.ChromaticAtmosphere(mono_PSF, base_wavelength=500,
+    PSF = galsim.ChromaticAtmosphere(mono_PSF, base_wavelength=500, 
```
```suggestion
    PSF = galsim.ChromaticAtmosphere(mono_PSF, base_wavelength=500,
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -471,14 +517,22 @@ def test_trapz_basic():
     np.testing.assert_almost_equal(
-        result/expected_val, 1.0, decimal=6, verbose=True,
-        err_msg='Test of trapz() with points failed for f(x)=x^2 from 0 to 1')
+        result / expected_val,
+        1.0,
+        decimal=6,
+        verbose=True,
+        err_msg="Test of trapz() with points failed for f(x)=x^2 from 0 to 1",
+    )
```
```suggestion
        result / expected_val, 1.0, decimal=6, verbose=True,
        err_msg='Test of trapz() with points failed for f(x)=x^2 from 0 to 1')
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -471,14 +517,22 @@ def test_trapz_basic():
     np.testing.assert_almost_equal(
-        result/expected_val, 1.0, decimal=6, verbose=True,
-        err_msg='Simple test of trapz() method failed for f(x)=x^2 from 0 to 1')
+        result / expected_val,
+        1.0,
+        decimal=6,
+        verbose=True,
+        err_msg="Simple test of trapz() method failed for f(x)=x^2 from 0 to 1",
+    )
```
```suggestion
        result/expected_val, 1.0, decimal=6, verbose=True,
        err_msg='Simple test of trapz() method failed for f(x)=x^2 from 0 to 1')
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -444,7 +488,9 @@ def test_surface_ops():
     assigner.rng.reset(rng)
     sampler.rng.reset(rng)
-    _, photons2 = check_dep(obj.drawPhot, image=im.copy(), surface_ops=[sampler, assigner], rng=rng)
+    _, photons2 = check_dep(
+        obj.drawPhot, image=im.copy(), surface_ops=[sampler, assigner], rng=rng
+    )
```
```suggestion
    _, photons2 = check_dep(obj.drawPhot, image=im.copy(), surface_ops=[sampler, assigner], rng=rng)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -432,7 +476,7 @@ def test_surface_ops():
     obj = galsim.Gaussian(flux=353, sigma=0.3)
     im = galsim.Image(63,63, scale=1)
-    check_dep(obj.drawImage, im, method='phot', surface_ops=[sampler, assigner], rng=rng,
+    check_dep(obj.drawImage, im, method="phot", surface_ops=[sampler, assigner], rng=rng,
```
```suggestion
    check_dep(obj.drawImage, im, method='phot', surface_ops=[sampler, assigner], rng=rng,
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -411,9 +449,15 @@ def test_roman_psfs():
-        check_dep(galsim.roman.getPSF, SCA=use_sca, bandpass='Z087',
-                            approximate_struts=False, high_accuracy=True,
-                            wavelength='Z087')
+        check_dep(
+            galsim.roman.getPSF,
+            SCA=use_sca,
+            bandpass="Z087",
+            approximate_struts=False,
+            high_accuracy=True,
+            wavelength="Z087",
+        )
+
```
```suggestion
        check_dep(galsim.roman.getPSF, SCA=use_sca, bandpass='Z087',
                            approximate_struts=False, high_accuracy=True,
                            wavelength='Z087')
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -329,8 +361,12 @@ def test_withOrigin():
         np.testing.assert_almost_equal(
-                world_pos2.distanceTo(world_pos1) / galsim.arcsec, 0, 7,
-                'shiftOrigin(new_origin) returned wrong world position')
+            world_pos2.distanceTo(world_pos1) / galsim.arcsec,
+            0,
+            7,
+            "shiftOrigin(new_origin) returned wrong world position",
+        )
+
```
```suggestion
                world_pos2.distanceTo(world_pos1) / galsim.arcsec, 0, 7,
                'shiftOrigin(new_origin) returned wrong world position')
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -477,7 +511,7 @@ def test_trapz_basic():
     expected_val = 1.**3./3.
     np.testing.assert_almost_equal(
-        result/expected_val, 1.0, decimal=6, verbose=True,
+        result / expected_val, 1.0, decimal=6, verbose=True,
```
```suggestion
        result/expected_val, 1.0, decimal=6, verbose=True,
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -244,8 +245,12 @@ def test_randwalk_config():
 
-        assert rw.input_half_light_radius==rwc.input_half_light_radius,\
-            "expected hlr==%g, got %g" % (rw.input_half_light_radius, rw.input_half_light_radius)
+        assert (
+            rw.input_half_light_radius == rwc.input_half_light_radius
+        ), "expected hlr==%g, got %g" % (
+            rw.input_half_light_radius,
+            rw.input_half_light_radius,
+        )
```
```suggestion
        assert rw.input_half_light_radius==rwc.input_half_light_radius,\
            "expected hlr==%g, got %g" % (rw.input_half_light_radius, rw.input_half_light_radius)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -261,14 +266,21 @@ def test_withOrigin():
-                                   xfunc = lambda u,v: u / scale, yfunc = lambda u,v: v / scale),
-                 galsim.UVFunction(ufunc='0.2*x + 0.03*y', vfunc='0.01*x + 0.2*y'),
-               ]
+    wcs_list = [
+        galsim.OffsetWCS(0.3, galsim.PositionD(1, 1), galsim.PositionD(10, 23)),
+        galsim.OffsetShearWCS(
+            0.23, galsim.Shear(g1=0.1, g2=0.3), galsim.PositionD(12, 43)
+        ),
+        galsim.AffineTransform(0.01, 0.26, -0.26, 0.02, galsim.PositionD(12, 43)),
+        galsim.UVFunction(ufunc=lambda x, y: 0.2 * x, vfunc=lambda x, y: 0.2 * y),
+        galsim.UVFunction(
+            ufunc=lambda x, y: 0.2 * x,
+            vfunc=lambda x, y: 0.2 * y,
+            xfunc=lambda u, v: u / scale,
+            yfunc=lambda u, v: v / scale,
+        ),
+        galsim.UVFunction(ufunc="0.2*x + 0.03*y", vfunc="0.01*x + 0.2*y"),
+    ]
```
```suggestion
    wcs_list = [ galsim.OffsetWCS(0.3, galsim.PositionD(1,1), galsim.PositionD(10,23)),
                 galsim.OffsetShearWCS(0.23, galsim.Shear(g1=0.1,g2=0.3), galsim.PositionD(12,43)),
                 galsim.AffineTransform(0.01,0.26,-0.26,0.02, galsim.PositionD(12,43)),
                 galsim.UVFunction(ufunc = lambda x,y: 0.2*x, vfunc = lambda x,y: 0.2*y),
                 galsim.UVFunction(ufunc = lambda x,y: 0.2*x, vfunc = lambda x,y: 0.2*y,
                                   xfunc = lambda u,v: u / scale, yfunc = lambda u,v: v / scale),
                 galsim.UVFunction(ufunc='0.2*x + 0.03*y', vfunc='0.01*x + 0.2*y'),
               ]
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -283,39 +295,59 @@ def test_withOrigin():
-            wcs2 = wcs.local(wcs.origin, color=color).withOrigin(wcs.origin, wcs.world_origin)
-            assert wcs == wcs2, name+' not equal after wcs.local().withOrigin(origin,world_origin)'
-        world_pos1 = wcs.toWorld(galsim.PositionD(0,0), color=color)
+                assert wcs == wcs2, (
+                    name + " is not equal after wcs.local().withOrigin(origin)"
+                )
+            wcs2 = wcs.local(wcs.origin, color=color).withOrigin(
+                wcs.origin, wcs.world_origin
+            )
+            assert wcs == wcs2, (
+                name + " not equal after wcs.local().withOrigin(origin,world_origin)"
+            )
+        world_pos1 = wcs.toWorld(galsim.PositionD(0, 0), color=color)
```
```suggestion
                assert wcs == wcs2, name+' is not equal after wcs.local().withOrigin(origin)'
            wcs2 = wcs.local(wcs.origin, color=color).withOrigin(wcs.origin, wcs.world_origin)
            assert wcs == wcs2, name+' not equal after wcs.local().withOrigin(origin,world_origin)'
        world_pos1 = wcs.toWorld(galsim.PositionD(0,0), color=color)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -283,39 +295,59 @@ def test_withOrigin():
         np.testing.assert_almost_equal(
-                world_pos2.x, world_pos1.x, 7,
-                'withOrigin(new_origin) returned wrong world position')
+            world_pos2.x,
+            world_pos1.x,
+            7,
+            "withOrigin(new_origin) returned wrong world position",
+        )
```
```suggestion
                world_pos2.x, world_pos1.x, 7,
                'withOrigin(new_origin) returned wrong world position')
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -283,39 +295,59 @@ def test_withOrigin():
         np.testing.assert_almost_equal(
-                world_pos2.y, world_pos1.y, 7,
-                'withOrigin(new_origin) returned wrong world position')
+            world_pos2.y,
+            world_pos1.y,
+            7,
+            "withOrigin(new_origin) returned wrong world position",
+        )
```
```suggestion
                world_pos2.y, world_pos1.y, 7,
                'withOrigin(new_origin) returned wrong world position')
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -283,39 +295,59 @@ def test_withOrigin():
         np.testing.assert_almost_equal(
-                world_pos3.x, new_world_origin.x, 7,
-                'withOrigin(new_origin, new_world_origin) returned wrong position')
+            world_pos3.x,
+            new_world_origin.x,
+            7,
+            "withOrigin(new_origin, new_world_origin) returned wrong position",
+        )
```
```suggestion
                world_pos3.x, new_world_origin.x, 7,
                'withOrigin(new_origin, new_world_origin) returned wrong position')
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -283,39 +295,59 @@ def test_withOrigin():
         np.testing.assert_almost_equal(
-                world_pos3.y, new_world_origin.y, 7,
-                'withOrigin(new_origin, new_world_origin) returned wrong position')
+            world_pos3.y,
+            new_world_origin.y,
+            7,
+            "withOrigin(new_origin, new_world_origin) returned wrong position",
+        )
```
```suggestion
                world_pos3.y, new_world_origin.y, 7,
                'withOrigin(new_origin, new_world_origin) returned wrong position')
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -309,13 +310,15 @@ def test_withOrigin():
-                 galsim.GSFitsWCS('tpv.fits', dir='fits_files'),
-                 galsim.FitsWCS('sipsample.fits', dir='fits_files'),
-               ]
+    radec = lambda x, y: center.deproject_rad(
+        cubic_u(x, y) * 0.2, cubic_v(x, y) * 0.2, projection="lambert"
+    )
+    wcs_list = [
+        galsim.RaDecFunction(radec),
+        galsim.AstropyWCS("1904-66_TAN.fits", dir="fits_files"),
+        galsim.GSFitsWCS("tpv.fits", dir="fits_files"),
+        galsim.FitsWCS("sipsample.fits", dir="fits_files"),
+    ]
```
```suggestion
    radec = lambda x,y: center.deproject_rad(cubic_u(x,y)*0.2, cubic_v(x,y)*0.2,
                                             projection='lambert')
    wcs_list = [ galsim.RaDecFunction(radec),
                 galsim.AstropyWCS('1904-66_TAN.fits', dir='fits_files'),
                 galsim.GSFitsWCS('tpv.fits', dir='fits_files'),
                 galsim.FitsWCS('sipsample.fits', dir='fits_files'),
               ]
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -389,17 +392,19 @@ def test_roman_psfs():
-         { 'pupil_bin':4, 'gsparams':galsim.GSParams(folding_threshold=2.e-3) }),
-        ({ 'approximate_struts':False, 'high_accuracy':False },
-         { 'pupil_bin':4 }),
+        ({"approximate_struts": True, "high_accuracy": False}, {"pupil_bin": 8}),
+        (
+            {"approximate_struts": True, "high_accuracy": True},
+            {"pupil_bin": 4, "gsparams": galsim.GSParams(folding_threshold=2.0e-3)},
+        ),
+        ({"approximate_struts": False, "high_accuracy": False}, {"pupil_bin": 4}),
```
```suggestion
        ({ 'approximate_struts':True, 'high_accuracy':False },
         { 'pupil_bin':8 }),
        ({ 'approximate_struts':True, 'high_accuracy':True },
         { 'pupil_bin':4, 'gsparams':galsim.GSParams(folding_threshold=2.e-3) }),
        ({ 'approximate_struts':False, 'high_accuracy':False },
         { 'pupil_bin':4 }),
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -389,17 +392,19 @@ def test_roman_psfs():
         test_kwargs.append(
-            ({ 'approximate_struts':False, 'high_accuracy':True },
-            { 'pupil_bin':1, 'gsparams':galsim.GSParams(folding_threshold=2.e-3) }),
+            (
+                {"approximate_struts": False, "high_accuracy": True},
+                {"pupil_bin": 1, "gsparams": galsim.GSParams(folding_threshold=2.0e-3)},
+            ),
```
```suggestion
            ({ 'approximate_struts':False, 'high_accuracy':True },
            { 'pupil_bin':1, 'gsparams':galsim.GSParams(folding_threshold=2.e-3) }),
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -49,36 +49,37 @@ def test_phase_psf():
-    l11 = check_dep(galsim.Interpolant.from_name, 'lanczos11', tol=1.e-2)
-    assert l11.gsparams.kvalue_accuracy == 1.e-2
-    assert check_dep(getattr, l11, 'tol') == l11.gsparams.kvalue_accuracy
+    d = check_dep(galsim.Delta, tol=1.0e-2)
+    assert d.gsparams.kvalue_accuracy == 1.0e-2
+    assert check_dep(getattr, d, "tol") == d.gsparams.kvalue_accuracy
+    n = check_dep(galsim.Nearest, tol=1.0e-2)
+    assert n.gsparams.kvalue_accuracy == 1.0e-2
+    assert check_dep(getattr, n, "tol") == n.gsparams.kvalue_accuracy
+    s = check_dep(galsim.SincInterpolant, tol=1.0e-2)
+    assert s.gsparams.kvalue_accuracy == 1.0e-2
+    assert check_dep(getattr, s, "tol") == s.gsparams.kvalue_accuracy
+    l = check_dep(galsim.Linear, tol=1.0e-2)
+    assert l.gsparams.kvalue_accuracy == 1.0e-2
+    assert check_dep(getattr, l, "tol") == l.gsparams.kvalue_accuracy
+    c = check_dep(galsim.Cubic, tol=1.0e-2)
+    assert c.gsparams.kvalue_accuracy == 1.0e-2
+    assert check_dep(getattr, c, "tol") == c.gsparams.kvalue_accuracy
+    q = check_dep(galsim.Quintic, tol=1.0e-2)
+    assert q.gsparams.kvalue_accuracy == 1.0e-2
+    assert check_dep(getattr, q, "tol") == q.gsparams.kvalue_accuracy
+    l3 = check_dep(galsim.Lanczos, 3, tol=1.0e-2)
+    assert l3.gsparams.kvalue_accuracy == 1.0e-2
+    assert check_dep(getattr, l3, "tol") == l3.gsparams.kvalue_accuracy
+    ldc = check_dep(galsim.Lanczos, 3, False, tol=1.0e-2)
+    assert ldc.gsparams.kvalue_accuracy == 1.0e-2
+    assert check_dep(getattr, ldc, "tol") == ldc.gsparams.kvalue_accuracy
+    l8 = check_dep(galsim.Lanczos, 8, tol=1.0e-2)
+    assert l8.gsparams.kvalue_accuracy == 1.0e-2
+    assert check_dep(getattr, l8, "tol") == l8.gsparams.kvalue_accuracy
+    l11 = check_dep(galsim.Interpolant.from_name, "lanczos11", tol=1.0e-2)
+    assert l11.gsparams.kvalue_accuracy == 1.0e-2
+    assert check_dep(getattr, l11, "tol") == l11.gsparams.kvalue_accuracy
+
```
```suggestion
    d = check_dep(galsim.Delta, tol=1.e-2)
    assert d.gsparams.kvalue_accuracy == 1.e-2
    assert check_dep(getattr, d, 'tol') == d.gsparams.kvalue_accuracy
    n = check_dep(galsim.Nearest, tol=1.e-2)
    assert n.gsparams.kvalue_accuracy == 1.e-2
    assert check_dep(getattr, n, 'tol') == n.gsparams.kvalue_accuracy
    s = check_dep(galsim.SincInterpolant, tol=1.e-2)
    assert s.gsparams.kvalue_accuracy == 1.e-2
    assert check_dep(getattr, s, 'tol') == s.gsparams.kvalue_accuracy
    l = check_dep(galsim.Linear, tol=1.e-2)
    assert l.gsparams.kvalue_accuracy == 1.e-2
    assert check_dep(getattr, l, 'tol') == l.gsparams.kvalue_accuracy
    c = check_dep(galsim.Cubic, tol=1.e-2)
    assert c.gsparams.kvalue_accuracy == 1.e-2
    assert check_dep(getattr, c, 'tol') == c.gsparams.kvalue_accuracy
    q = check_dep(galsim.Quintic, tol=1.e-2)
    assert q.gsparams.kvalue_accuracy == 1.e-2
    assert check_dep(getattr, q, 'tol') == q.gsparams.kvalue_accuracy
    l3 = check_dep(galsim.Lanczos, 3, tol=1.e-2)
    assert l3.gsparams.kvalue_accuracy == 1.e-2
    assert check_dep(getattr, l3, 'tol') == l3.gsparams.kvalue_accuracy
    ldc = check_dep(galsim.Lanczos, 3, False, tol=1.e-2)
    assert ldc.gsparams.kvalue_accuracy == 1.e-2
    assert check_dep(getattr, ldc, 'tol') == ldc.gsparams.kvalue_accuracy
    l8 = check_dep(galsim.Lanczos, 8, tol=1.e-2)
    assert l8.gsparams.kvalue_accuracy == 1.e-2
    assert check_dep(getattr, l8, 'tol') == l8.gsparams.kvalue_accuracy
    l11 = check_dep(galsim.Interpolant.from_name, 'lanczos11', tol=1.e-2)
    assert l11.gsparams.kvalue_accuracy == 1.e-2
    assert check_dep(getattr, l11, 'tol') == l11.gsparams.kvalue_accuracy
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -540,96 +540,130 @@ def test_photon_array_depr():
-    dxdz = check_dep(getattr, photon_array, 'dxdz')
-    assert photon_array.hasAllocatedAngles()
-    assert photon_array.hasAllocatedAngles()
+    # jax-galsim always sets these additional properties
+    # dxdz = check_dep(getattr, photon_array, 'dxdz')
+    # however jax-galsim sets them to NaN so they are not allocated
+    assert not photon_array.hasAllocatedAngles()
```
```suggestion
    if hasattr(galsim, "_galsim"):
        dxdz = check_dep(getattr, photon_array, 'dxdz')
        assert photon_array.hasAllocatedAngles()
    else:
        # jax-galsim always sets these additional properties
        # dxdz = check_dep(getattr, photon_array, 'dxdz')
        # however jax-galsim sets them to NaN so they are not allocated
        assert not photon_array.hasAllocatedAngles()
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -540,96 +540,130 @@ def test_photon_array_depr():
 
-    dydz = photon_array.dydz  # Allowed now.
-    dydz[:] = 0.59
+    # JAX-Galsim does not allow by reference setting - changed this
```
```suggestion
    assert hasattr(photon_array, "dydz")
    # JAX-Galsim does not allow by reference setting - changed this
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -540,96 +540,130 @@ def test_photon_array_depr():
 
-    wave = check_dep(getattr, photon_array, 'wavelength')
-    assert photon_array.hasAllocatedWavelengths()
+    # jax-galsim always sets these additional properties
+    # wave = check_dep(getattr, photon_array, 'wavelength')
+    # however jax-galsim sets them to NaN so they are not allocated
+    assert not photon_array.hasAllocatedWavelengths()
```
```suggestion
    if hasattr(galsim, "_galsim"):
        wave = check_dep(getattr, photon_array, 'wavelength')
        assert photon_array.hasAllocatedWavelengths()
    else:
        # jax-galsim always sets these additional properties
        # wave = check_dep(getattr, photon_array, 'wavelength')
        # however jax-galsim sets them to NaN so they are not allocated
        assert not photon_array.hasAllocatedWavelengths()
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -540,96 +540,139 @@ def test_photon_array_depr():
-    u = check_dep(getattr, photon_array, 'pupil_u')
-    assert photon_array.hasAllocatedPupil()
-    assert len(photon_array.pupil_u) == nphotons
+    # jax-galsim always sets these additional properties
+    # u = check_dep(getattr, photon_array, "pupil_u")
+    # however jax-galsim sets them to NaN so they are not allocated
+    assert not photon_array.hasAllocatedPupil()
```
```suggestion
    if hasattr(galsim, "_galsim"):
        u = check_dep(getattr, photon_array, 'pupil_u')
        assert photon_array.hasAllocatedPupil()
        assert len(photon_array.pupil_u) == nphotons
    else:
        # jax-galsim always sets these additional properties
        # u = check_dep(getattr, photon_array, "pupil_u")
        # however jax-galsim sets them to NaN so they are not allocated
        assert not photon_array.hasAllocatedPupil()
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -540,96 +540,144 @@ def test_photon_array_depr():
     np.testing.assert_array_equal(photon_array.pupil_v, 0.0)
-    v = photon_array.pupil_v
-    v[:] = 10.0
+    # JAX-Galsim does not allow by reference setting - changed this
```
```suggestion
    assert hasattr(photon_array, "pupil_v")
    # JAX-Galsim does not allow by reference setting - changed this
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -540,96 +540,144 @@ def test_photon_array_depr():
 
-    t = check_dep(getattr, photon_array, 'time')
-    assert photon_array.hasAllocatedTimes()
+    # jax-galsim always sets these additional properties
+    # t = check_dep(getattr, photon_array, "time")
+    # however jax-galsim sets them to NaN so they are not allocated
+    assert not photon_array.hasAllocatedTimes()
     assert len(photon_array.time) == nphotons
```
```suggestion
    if hasattr(galsim, "_galsim"):
        t = check_dep(getattr, photon_array, 'time')
        assert photon_array.hasAllocatedTimes()    
    else:
        # jax-galsim always sets these additional properties
        # t = check_dep(getattr, photon_array, "time")
        # however jax-galsim sets them to NaN so they are not allocated
        assert not photon_array.hasAllocatedTimes()
        assert len(photon_array.time) == nphotons
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -540,96 +540,149 @@ def test_photon_array_depr():
+    # jax-galsim is allocated now
     assert photon_array.hasAllocatedTimes()
-    assert len(photon_array.time) == nphotons
     np.testing.assert_array_equal(photon_array.time, 0.0)
```
```suggestion
    assert len(photon_array.time) == nphotons
    np.testing.assert_array_equal(photon_array.time, 0.0)
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -540,96 +540,149 @@ def test_photon_array_depr():
-    dydz = check_dep(getattr, photon_array, 'dydz')
-    assert photon_array.hasAllocatedAngles()
-    assert photon_array.hasAllocatedAngles()
+    # jax-galsim always sets these additional properties
+    # dydz = check_dep(getattr, photon_array, "dydz")
+    # however jax-galsim sets them to NaN so they are not allocated
+    assert not photon_array.hasAllocatedAngles()
```
```suggestion
    if hasattr(galsim, "_galsim"):
        dydz = check_dep(getattr, photon_array, 'dydz')
        assert photon_array.hasAllocatedAngles()
    else:
        # jax-galsim always sets these additional properties
        # dydz = check_dep(getattr, photon_array, "dydz")
        # however jax-galsim sets them to NaN so they are not allocated
        assert not photon_array.hasAllocatedAngles()
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -540,96 +540,149 @@ def test_photon_array_depr():
 
-    dxdz = photon_array.dxdz  # Allowed now.
-    dxdz[:] = 0.17
+    # JAX-Galsim does not allow by reference setting - changed this
```
```suggestion
    assert hasattr(photon_array, "dxdz")
    # JAX-Galsim does not allow by reference setting - changed this
```

beckermr committed:
Apply suggestions from code review

beckermr commented on tests/test_deprecated.py:
```diff
@@ -540,96 +540,154 @@ def test_photon_array_depr():
-    v = check_dep(getattr, photon_array, 'pupil_v')
-    assert photon_array.hasAllocatedPupil()
-    assert len(photon_array.pupil_u) == nphotons
+    # jax-galsim always sets these additional properties
+    # v = check_dep(getattr, photon_array, "pupil_v")
+    # however jax-galsim sets them to NaN so they are not allocated
+    assert not photon_array.hasAllocatedPupil()
```
```suggestion
    if hasattr(galsim, "_galsim"):
        v = check_dep(getattr, photon_array, 'pupil_v')
        assert photon_array.hasAllocatedPupil()
        assert len(photon_array.pupil_u) == nphotons    
    else:
        # jax-galsim always sets these additional properties
        # v = check_dep(getattr, photon_array, "pupil_v")
        # however jax-galsim sets them to NaN so they are not allocated
        assert not photon_array.hasAllocatedPupil()
```

beckermr committed:
Apply suggestions from code review

beckermr committed:
these are not supported at all in jax-galsim

beckermr committed:
WIP update test for jax-galsim memory use

beckermr committed:
TST remove extra B

beckermr committed:
TST less memory

beckermr committed:
TST less memory

beckermr committed:
TST less memory

beckermr committed:
TST less memory

beckermr committed:
TST less memory

beckermr committed:
TST less memory

beckermr committed:
TST cancel concurrent tests

beckermr committed:
TST cancel concurrent tests

beckermr committed:
TST fix fft tests

beckermr committed:
test: update tests for jax galsim

beckermr committed:
test: lower precision for some interp image tests

beckermr committed:
Revert "test: lower precision for some interp image tests"

This reverts commit ff175e1d6148c9bebf1a9248f841e88c73212656.

beckermr committed:
Merge branch 'main' into fix_tests

beckermr committed:
fix: missing import

beckermr committed:
test: more missing imports and tests

beckermr committed:
test: rename the imports

beckermr committed:
feat: add function to extract the galsim backend

beckermr committed:
fix: start to use a proper helper function

beckermr committed:
test: convert more to use the helper functions

beckermr commented on tests/galsim_test_helpers.py:
```diff
@@ -206,6 +260,30 @@ def check_basic_k(prof, name):
+    assert (
+        isinstance(val, float)
+        or (
+            (not hasattr(galsim, "_galsim"))
```
```suggestion
            is_jax_galsim()
```

beckermr commented on tests/galsim_test_helpers.py:
```diff
@@ -206,6 +260,30 @@ def check_basic_k(prof, name):
+    assert (
+        isinstance(val, int)
+        or (
+            (not hasattr(galsim, "_galsim"))
```
```suggestion
            is_jax_galsim()
```

beckermr commented on tests/galsim_test_helpers.py:
```diff
@@ -298,6 +376,8 @@ def do_shoot(prof, img, name):
     print('nphot = ',nphot)
     img2 = img.copy()
 
+    rtol *= 3
```
```suggestion
    if is_jax_galsim():
        rtol *= 3
```

beckermr committed:
test: fix test helpers file for new functions

beckermr commented on tests/test_celestial_galsim.py:
```diff
@@ -122,11 +123,17 @@ def test_angle():
     # Check invalid constructors
     assert_raises(TypeError,galsim.AngleUnit, galsim.degrees)
-    assert_raises(ValueError,galsim.AngleUnit, 'spam')
+    try:
+        assert_raises(ValueError,galsim.AngleUnit, 'spam')
+    except Exception:
+        assert_raises(TypeError,galsim.AngleUnit, 'spam')
```
```suggestion
    assert_raises((ValueError, TypeError), galsim.AngleUnit, 'spam')
```

beckermr commented on tests/test_celestial_galsim.py:
```diff
@@ -122,11 +123,17 @@ def test_angle():
     assert_raises(TypeError,galsim.Angle, 3.4)
     assert_raises(TypeError,galsim.Angle, theta1, galsim.degrees)
-    assert_raises(ValueError,galsim.Angle, 'spam', galsim.degrees)
+    try:
+        assert_raises(ValueError,galsim.Angle, 'spam', galsim.degrees)
+    except Exception:
+        assert_raises(TypeError,galsim.Angle, 'spam', galsim.degrees)
```
```suggestion
    assert_raises((ValueError, TypeError), galsim.AngleUnit, 'spam'))
```

beckermr commented on tests/test_celestial_galsim.py:
```diff
@@ -343,9 +350,17 @@ def test_projection():
-    assert c0 == center
+    np.testing.assert_allclose(
+        c0.rad,
+        center.rad,
```
```suggestion
        center.rad,
        rtol=0,
        atol=1e-16,        
```

beckermr commented on tests/test_celestial_galsim.py:
```diff
@@ -398,9 +413,17 @@ def test_projection():
-    assert c0 == center
+    np.testing.assert_allclose(
+        c0.rad,
+        center.rad,
```
```suggestion
        center.rad,
        rtol=0,
        atol=1e-16,
```

beckermr commented on tests/test_celestial_galsim.py:
```diff
@@ -456,9 +479,17 @@ def test_projection():
-    assert c0 == center
+    np.testing.assert_allclose(
+        c0.rad,
+        center.rad,
```
```suggestion
        center.rad,
        rtol=0,
        atol=1e-16,
```

beckermr commented on tests/test_celestial_galsim.py:
```diff
@@ -510,9 +541,17 @@ def test_projection():
-    assert c0 == center
+    np.testing.assert_allclose(
+        c0.rad,
+        center.rad,
```
```suggestion
        center.rad,
        rtol=0,
        atol=1e-16,
```

beckermr committed:
test: clean up celestial tests

beckermr commented on tests/test_celestial_galsim.py:
```diff
@@ -122,11 +123,11 @@ def test_angle():
     assert_raises(TypeError,galsim.Angle, 3.4)
     assert_raises(TypeError,galsim.Angle, theta1, galsim.degrees)
-    assert_raises(ValueError,galsim.Angle, 'spam', galsim.degrees)
+    assert_raises((ValueError, TypeError), galsim.AngleUnit, 'spam'))
```
```suggestion
    assert_raises((ValueError, TypeError), galsim.Angle, 'spam'))
```

beckermr committed:
fix: need to test Angle

beckermr commented on tests/test_celestial_galsim.py:
```diff
@@ -122,11 +123,11 @@ def test_angle():
     assert_raises(TypeError,galsim.Angle, 3.4)
     assert_raises(TypeError,galsim.Angle, theta1, galsim.degrees)
-    assert_raises(ValueError,galsim.Angle, 'spam', galsim.degrees)
+    assert_raises((ValueError, TypeError), galsim.Angle, 'spam'))
```
```suggestion
    assert_raises((ValueError, TypeError), galsim.Angle, 'spam')
```

beckermr committed:
fix: bug in parens

beckermr commented on tests/test_optics.py:
```diff
@@ -745,7 +745,11 @@ def test_OpticalPSF_pupil_plane_size():
     x = y = np.arange(512) - 256
     y, x = np.meshgrid(y, x)
-    im.array[x**2+y**2 < 230**2] = 1.0
+    if hasattr(galsim, "_galsim"):
+        im.array[x**2+y**2 < 230**2] = 1.0
+    else:
+        # no refs in jax-galsim
+        im._array = im.array.at[x**2+y**2 < 230**2].set(1.0)
```
```suggestion
    if is_jax_galsim():
        # no refs in jax-galsim
        im._array = im.array.at[x**2+y**2 < 230**2].set(1.0)
    else:
        im.array[x**2+y**2 < 230**2] = 1.0
```

beckermr committed:
test: clean up test_optics

beckermr committed:
tests: always default to running tests for other backends

beckermr commented on tests/test_exponential.py:
```diff
@@ -226,7 +226,9 @@ def test_exponential_shoot():
     assert np.isclose(im.array.sum(), obj.flux)
     photons2 = obj.makePhot(poisson_flux=False, rng=rng)
-    assert photons2 == photons, "Exponential makePhot not equivalent to drawPhot"
+    np.testing.assert_allclose(photons2.x, photons.x)
+    np.testing.assert_allclose(photons2.y, photons.y)
+    np.testing.assert_allclose(photons2.flux, photons.flux)
```
```suggestion
    if is_jax_galsim():
        np.testing.assert_allclose(photons2.x, photons.x)
        np.testing.assert_allclose(photons2.y, photons.y)
        np.testing.assert_allclose(photons2.flux, photons.flux)
    else:
        assert photons2 == photons, "Exponential makePhot not equivalent to drawPhot"
```

beckermr committed:
test: clean up tests for exp w/ phots

beckermr commented on tests/test_moffat.py:
```diff
@@ -135,36 +136,42 @@ def test_moffat_properties():
-    np.testing.assert_almost_equal(psf.kValue(cen), (1+0j) * test_flux)
-    np.testing.assert_almost_equal(psf.flux, test_flux)
-    np.testing.assert_almost_equal(psf.xValue(cen), psf.max_sb)
+    if hasattr(galsim, "_galsim"):
+        np.testing.assert_array_almost_equal(psf.maxk, 11.634597424960159)
+    else:
+        np.testing.assert_allclose(psf.maxk, 11.634597424960159, atol=0, rtol=0.2)
```
```suggestion
    if is_jax_galsim():
        np.testing.assert_array_almost_equal(psf.maxk, 11.634597424960159, atol=0, rtol=0.2)
    else:
        np.testing.assert_array_almost_equal(psf.maxk, 11.634597424960159)
```

beckermr commented on tests/test_moffat.py:
```diff
@@ -135,36 +136,42 @@ def test_moffat_properties():
-    np.testing.assert_almost_equal(psf.kValue(cen), (1+0j) * test_flux)
-    np.testing.assert_almost_equal(psf.flux, test_flux)
-    np.testing.assert_almost_equal(psf.xValue(cen), psf.max_sb)
+    if hasattr(galsim, "_galsim"):
+        np.testing.assert_array_almost_equal(psf.maxk, 11.634597424960159)
+    else:
+        np.testing.assert_allclose(psf.maxk, 11.634597424960159, atol=0, rtol=0.2)
```
```suggestion
    if is_jax_galsim():
        np.testing.assert_array_almost_equal(psf.maxk, 11.634597424960159, atol=0, rtol=0.2)
    else:
        np.testing.assert_array_almost_equal(psf.maxk, 11.634597424960159)
```

beckermr commented on tests/test_moffat.py:
```diff
@@ -190,13 +197,16 @@ def test_moffat_maxk():
         for thresh in threshs:
             psf = psf.withGSParams(maxk_threshold=thresh)
-            rtol = 1.e-7 if psf.trunc == 0 else 3.e-3
+            if hasattr(galsim, "_galsim"):
+                rtol = 1.e-7 if psf.trunc == 0 else 3.e-3
+            else:
+                rtol = 5e-3
```
```suggestion
            if is_jax_galsim():
                rtol = 5e-3
            else:
                rtol = 1.e-7 if psf.trunc == 0 else 3.e-3
```

beckermr committed:
test: clean up tests for moffat

beckermr committed:
fix: testing functions do not have atol/rtol

beckermr committed:
test: fix first part of deprecation test file

beckermr commented on tests/test_deprecated.py:
```diff
@@ -540,96 +543,180 @@ def test_photon_array_depr():
+    if is_jax_galsim():
+        # JAX-Galsim does not allow by reference setting - changed this
+        # to make tests below run
+        photon_array.wavelength = 500.0
```
```suggestion
        photon_array.wavelength = 500.0
        # jax-galsim is allocated now
        assert photon_array.hasAllocatedWavelengths()
```

beckermr committed:
test: better test here

beckermr commented on tests/test_deprecated.py:
```diff
@@ -540,96 +543,182 @@ def test_photon_array_depr():
+        assert not photon_array.hasAllocatedTimes()
+        # jax-galsim needs to set 0
+        photon_array.time = 0.0
+        # jax-galsim is allocated now
```
```suggestion
        # jax-galsim is allocated now
        assert photon_array.hasAllocatedTimes()
```

beckermr committed:
test: check allocated

beckermr committed:
test: clean up more test stuff

beckermr committed:
test: clean up more test stuff

beckermr commented on tests/test_deprecated.py:
```diff
@@ -540,96 +543,199 @@ def test_photon_array_depr():
+    if is_jax_galsim():
+        # JAX-Galsim does not allow by reference setting - changed this
+        # to make tests below run
+        photon_array.dydz = 0.59
```
```suggestion
        photon_array.dydz = 0.59
        # non-nan means allocated for jax-galsim
        assert photon_array.hasAllocatedAngles()
```

beckermr commented on tests/test_deprecated.py:
```diff
@@ -540,96 +543,199 @@ def test_photon_array_depr():
+    if is_jax_galsim():
+        # JAX-Galsim does not allow by reference setting - changed this
+        # to make tests below run
+        photon_array.pupil_v = 10.0
```
```suggestion
        photon_array.pupil_v = 10.0
        # jax-galsim is allocated now
        assert photon_array.hasAllocatedPupil()
```

beckermr committed:
test: get rest of test cases

beckermr committed:
test: clean up tests for photon arrays

beckermr committed:
test: clean up for interpolated images

beckermr committed:
test: clean up for noise tests

beckermr committed:
test: try using the olde value

beckermr committed:
test: clean up tests for draw

beckermr committed:
test: clean up tests for draw

beckermr committed:
test: clean up tests for draw

beckermr committed:
test: clean up tests for image

beckermr committed:
test: clean up tests for utils

beckermr committed:
test: clean up random tests

beckermr committed:
fix: jax_galsim does not raise

beckermr committed:
fix: wrong order of if statement

beckermr committed:
debug: figure out what is taking so long

beckermr committed:
debug: how many

beckermr committed:
debug: how fast

beckermr committed:
test: just do less for jax

beckermr committed:
test: just do less for jax

beckermr committed:
test: it is an array in jax

beckermr committed:
test: it is an array in jax

beckermr committed:
test: it is an array in jax

beckermr committed:
test: it is an array in jax

rmjarvis commented on tests/test_shear.py:
```diff
@@ -176,17 +176,23 @@ def test_shear_initialization():
+        assert_raises(galsim.GalSimRangeError,galsim.Shear,g1=0.9,g2=0.6)
+        assert_raises(galsim.GalSimRangeError,galsim.Shear,e=-1.3,beta=0.*galsim.radians)
+        assert_raises(galsim.GalSimRangeError,galsim.Shear,e=1.3,beta=0.*galsim.radians)
+        assert_raises(galsim.GalSimRangeError,galsim.Shear,e1=0.7,e2=0.9)
```
What does JaxGalSim do with these inputs?  They don't define valid Shear values.

beckermr replied:
It creates NaNs. 

```
In [1]: import jax_galsim

In [2]: s = jax_galsim.Shear(e1=0.7,e2=0.9)

In [3]: s
Out[3]: galsim.Shear(nan)

In [4]: s.e1
Out[4]: Array(nan, dtype=float32, weak_type=True)

In [5]: s.e2
Out[5]: Array(nan, dtype=float32, weak_type=True)

In [6]: s.g1
Out[6]: Array(nan, dtype=float32, weak_type=True)

In [7]: s.g2
Out[7]: Array(nan, dtype=float32, weak_type=True)

```

That is not as good as a raising an error, but is good enough. The NaNs will be noticed for sure and jax has decent utilities for tracking them down.

rmjarvis commented on tests/test_utilities.py:
```diff
@@ -252,7 +282,11 @@ def test_bounds():
-    assert_raises(ValueError, galsim.BoundsD, 11, 23, 17, "blue")
+    if is_jax_galsim():
+        # jax doesn't raise for this
+        pass
```
I think I can guess what it does with floats.  But I'm quite curious what it does with "blue".

beckermr replied:
It raises a type error. 

```
TypeError: unsupported operand type(s) for -: 'str' and 'int'
```

rmjarvis commented on tests/test_utilities.py:
```diff
@@ -373,15 +407,22 @@ def test_bounds():
+
+    if is_jax_galsim():
+        # jax doesn't raise for these things
+        pass
```
What does it do for an undefined Bounds if you ask for the center?

beckermr replied:
This now raises the correct error.

rmjarvis commented on tests/test_utilities.py:
```diff
@@ -501,15 +542,14 @@ def test_check_all_contiguous():
-    im_list, offsets = deInterleaveImage(img,8)
-    img1 = interleaveImages(im_list,8,offsets)
+    im_list, offsets = galsim.utilities.deInterleaveImage(img,8)
+    img1 = galsim.utilities.interleaveImages(im_list,8,offsets)
```
This is fine I guess, but I'm curious why this was required?

beckermr replied:
To run the galsim test suite with jax-galsim, we basically patch `galsim` top `jax_galsim` globally in all modules. The local import here bypassed that patching.

beckermr replied:
Or at least that is what I remember happening...

rmjarvis commented on tests/test_wcs.py:
```diff
@@ -499,7 +500,7 @@ def do_local_wcs(wcs, ufunc, vfunc, name):
     assert wcs == wcs2, name+' local() is not == the original'
     new_origin = galsim.PositionI(123,321)
-    wcs3 = wcs.withOrigin(new_origin)
+    wcs3 = wcs.shiftOrigin(new_origin)
```
For a LocalWCS, these are equivalent.  And IMO withOrigin is preferred, since the WCS doesn't have an origin yet.

beckermr replied:
xref: https://github.com/GalSim-developers/GalSim-for-JAX-GalSim-Testing/pull/13

rmjarvis commented on tests/test_zernike.py:
```diff
@@ -41,10 +40,10 @@ def test_Zernike_orthonormality():
     area = np.pi*R_outer**2
     for j1 in range(1, jmax+1):
-        Z1 = Zernike([0]*(j1+1)+[1], R_outer=R_outer)
+        Z1 = galsim.zernike.Zernike([0]*(j1+1)+[1], R_outer=R_outer)
```
Again, why add the extra boilerplate?

beckermr replied:
This change was related to the global patching of modules for running the tests with jax_galsim IIRC.

beckermr committed:
Merge branch 'main' into fix_tests

beckermr committed:
fix: make sure to export the pytest import

beckermr committed:
fix: make sure to export the new fixtures and functions

beckermr committed:
fix: make sure path to input test data is correct

EiffL committed:
updating values for random tests (#1344)

beckermr committed:
fix: use correct path to fits files

beckermr committed:
fix: use array comparison

beckermr committed:
fix: use correct path to other comp dirs

beckermr committed:
fix: jax-galsim is good on moffat maxk now

beckermr commented on tests/galsim_test_helpers.py:
```diff
@@ -298,6 +371,9 @@ def do_shoot(prof, img, name):
     img2 = img.copy()
 
+    if is_jax_galsim():
+        rtol *= 3
```
This was due to bug. Has now been fixed! Thanks for flagging!

beckermr commented on tests/test_interpolatedimage.py:
```diff
@@ -369,8 +369,12 @@ def test_interpolant():
+            np.testing.assert_allclose(ln.kval(x[12]), true_kval[12], rtol=3.0e-4, atol=3.0e-6)
+        else:
+            np.testing.assert_allclose(ln.kval(x), true_kval, rtol=1.e-4, atol=1.e-8)
+            assert np.isclose(ln.kval(x[12]), true_kval[12])
```
This was due to jax-galsim using a slightly less accurate interpolation method (akima vs natural cubic spline). This is now fixed.

beckermr commented on tests/test_moffat.py:
```diff
@@ -135,36 +136,42 @@ def test_moffat_properties():
-    np.testing.assert_almost_equal(psf.flux, test_flux)
-    np.testing.assert_almost_equal(psf.xValue(cen), psf.max_sb)
+    if is_jax_galsim():
+        np.testing.assert_allclose(psf.maxk, 11.634597424960159, atol=0, rtol=0.2)
```
We removed the `trunc` option in jax-galsim, so this change is no longer relevant.

beckermr commented on tests/test_random.py:
```diff
@@ -1927,32 +2168,49 @@ def test_permute():
+        a = galsim.BaseDeviate(seed="(0, 10)")
+        b = galsim.BaseDeviate(seed="(0, 11)")
+        assert repr(a) != repr(b)
+        assert a != b
```
So jax-galsim doesn't have DistDeviate implement and so doesn't pass the test below. jax-galsim RNG seeds are short and so includes them in the repr.

beckermr commented on tests/test_random.py:
```diff
@@ -982,16 +1164,20 @@ def test_poisson_zeromean():
-        p.generate_from_expectation(test_array)
+    # jax doesn't raise here
+    if is_jax_galsim():
+        pass
```
It returns -1 for every call. That's fun.

```
In [6]: pd()
Out[6]: Array(-1., dtype=float32)

In [7]: pd()
Out[7]: Array(-1., dtype=float32)

In [8]: pd()
Out[8]: Array(-1., dtype=float32)

In [9]: pd()
Out[9]: Array(-1., dtype=float32)
```

xref: https://github.com/GalSim-developers/JAX-GalSim/issues/237

beckermr commented on tests/test_celestial_galsim.py:
```diff
@@ -122,11 +123,11 @@ def test_angle():
     assert_raises(TypeError,galsim.Angle, 3.4)
     assert_raises(TypeError,galsim.Angle, theta1, galsim.degrees)
-    assert_raises(ValueError,galsim.Angle, 'spam', galsim.degrees)
+    assert_raises((ValueError, TypeError), galsim.Angle, 'spam')
```
fixed.

beckermr commented on tests/test_celestial_galsim.py:
```diff
@@ -343,9 +344,19 @@ def test_projection():
+        (0.0, 0.0),
+        rtol=0,
+        atol=1e-16,
+    )
```
The code here follows galsim/coord exactly. So I think this is due to the precise way in which jax does the computation. Under JIT, things can get reordered and simplified. I did not track this down further than that speculation.

beckermr commented on tests/galsim_test_helpers.py:
```diff
@@ -133,20 +170,32 @@ def check_basic_x(prof, name, approx_maxsb=False, scale=None):
-    np.testing.assert_almost_equal(neg_image.array/prof.flux, -image.array/prof.flux, 7,
-                                   '%s negative flux drawReal is not negative of +flux image'%name)
+    np.testing.assert_array_almost_equal(neg_image.array/prof.flux, -image.array/prof.flux, 7,
+                                         '%s negative flux drawReal is not negative of +flux image'%name)
```
They are not for jax arrays at least. It raises this error:

```
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

beckermr commented on tests/test_draw.py:
```diff
@@ -519,21 +527,29 @@ def test_drawKImage():
+        maxk_threshold = 1.e-4
+        N = 1174
+        Ns = 37
+
```
This is now fixed.

beckermr commented on tests/test_draw.py:
```diff
@@ -1270,14 +1322,22 @@ def test_fft():
+    else:
+        maxk_threshold = 1.e-4
+        N = 1174
+        Nfft = 1536
```
This is now fixed.

beckermr commented on tests/test_draw.py:
```diff
@@ -1524,8 +1584,12 @@ def test_np_fft():
-        array = np.around(array)
+    if is_jax_galsim():
+        # NOTE JAX doesn't round to the nearest int when drawing
+        pass
```
Interesting, this change is not needed when jax-galsim is working in float32 as opposed to float64.

beckermr replied:
Fixed here: https://github.com/GalSim-developers/JAX-GalSim/pull/238

beckermr commented on tests/test_image.py:
```diff
@@ -201,16 +205,32 @@ def test_Image_basic():
-                assert im2(x,y) == value
+                if is_jax_galsim():
+                    # no real views in jax
+                    assert im2(x,y) != value
```
That is a better default. I will make the change.

beckermr replied:
xref: https://github.com/GalSim-developers/JAX-GalSim/pull/239

beckermr commented on tests/test_draw.py:
```diff
@@ -1120,16 +1148,19 @@ def test_shoot():
-    with assert_warns(galsim.GalSimWarning):
-        psf.makePhot()
+    if is_jax_galsim():
+        pass
```
There are a few things happening here:

- Once a function is JITed, you cannot raise python-based warnings or errors. Some libraries like equinox have utilities to help with this, which would be an option.
- In order to use JIT or autodiff, JAX will trace the execution path of the function. The tracers are abstract things that hold basic pieces of information like array shapes, dtypes, etc, but do not hold a specific value for the array. So we cannot check their values and raise for disallowed values.
- As a result of the previous item, one cannot branch in a function based on a JAX array's value. This precludes many of the data checks.

We have an open issue for this here: https://github.com/GalSim-developers/JAX-GalSim/issues/237

beckermr commented on tests/test_image.py:
```diff
@@ -2415,7 +2470,11 @@ def test_Image_view():
+        # jax-galsim does not support views
+        assert im(11,19) != 20
+    else:
+        assert im(11,19) == 20
```
This test is skipped now due to views not being supported.

beckermr commented on tests/test_draw.py:
```diff
@@ -375,21 +376,28 @@ def test_drawImage():
-    assert_raises(ValueError, obj.drawImage, image=im10, exptime=0.)
-    assert_raises(ValueError, obj.drawImage, image=im10, exptime=-1.)
+    if is_jax_galsim():
+        pass
```
It is that we don't warn if they are invalid values.

beckermr commented on tests/test_deprecated.py:
```diff
@@ -34,9 +34,12 @@ def check_dep(f, *args, **kwargs):
+    else:
+        check_dep(galsim.GSParams, allowed_flux_variation=0.90)
+        check_dep(galsim.GSParams, range_division_for_extrema=50)
+        check_dep(galsim.GSParams, small_fraction_of_flux=1.e-6)
```
xref: https://github.com/GalSim-developers/GalSim-for-JAX-GalSim-Testing/pull/13

beckermr commented on tests/test_draw.py:
```diff
@@ -1219,9 +1259,15 @@ def test_fft():
+        else:
+            kim = xim.calculate_fft()
+            xim2 = kim.calculate_inverse_fft()
+            np.testing.assert_array_almost_equal(xim.array, xim2.array)
```
xref: https://github.com/GalSim-developers/GalSim-for-JAX-GalSim-Testing/pull/13

beckermr commented on tests/test_draw.py:
```diff
@@ -1253,9 +1299,15 @@ def test_fft():
+        else:
+            kim = xim.calculate_fft()
+            kim2 = xim2.calculate_fft()
+            np.testing.assert_array_almost_equal(kim.array, kim2.array)
```
xref: https://github.com/GalSim-developers/GalSim-for-JAX-GalSim-Testing/pull/13

beckermr commented on tests/test_interpolatedimage.py:
```diff
@@ -1845,33 +1872,39 @@ def test_depixelize():
-    np.testing.assert_array_equal(image.array, 0)
+    # this test only runs with real galsim
+    if is_jax_galsim():
+        pass
```
https://github.com/GalSim-developers/GalSim-for-JAX-GalSim-Testing/pull/13

beckermr commented on tests/test_photon_array.py:
```diff
@@ -181,30 +185,36 @@ def test_photon_array():
+    if is_jax_galsim():
+        pa1.flux = photon_array.flux[:50]
+    else:
+        pa1.flux[0:50] = photon_array.flux[:50]
```
xref: https://github.com/GalSim-developers/GalSim-for-JAX-GalSim-Testing/pull/13

beckermr commented on tests/test_moffat.py:
```diff
@@ -135,36 +136,42 @@ def test_moffat_properties():
+    np.testing.assert_array_almost_equal(psf.xValue(cen), 0.50654651638242509)
+    np.testing.assert_array_almost_equal(psf.kValue(cen), (1+0j) * test_flux)
+    np.testing.assert_array_almost_equal(psf.flux, test_flux)
+    np.testing.assert_array_almost_equal(psf.xValue(cen), psf.max_sb)
```
jax-galsim no longer supports truncated moffat profiles. so I removed this change. xref: https://github.com/GalSim-developers/GalSim-for-JAX-GalSim-Testing/pull/13

beckermr commented on tests/test_exponential.py:
```diff
@@ -226,7 +226,12 @@ def test_exponential_shoot():
+        np.testing.assert_allclose(photons2.y, photons.y)
+        np.testing.assert_allclose(photons2.flux, photons.flux)
+    else:
+        assert photons2 == photons, "Exponential makePhot not equivalent to drawPhot"
```
They are. Changing back: https://github.com/GalSim-developers/GalSim-for-JAX-GalSim-Testing/pull/14

beckermr commented on tests/test_photon_array.py:
```diff
@@ -60,12 +60,16 @@ def test_photon_array():
-    np.testing.assert_array_equal(photon_array.flux, 23.)
+    # jax does not support direct assignment
+    if is_jax_galsim():
+        pass
```
Yes. JAX itself raises

```
In [3]: photon_array.x[:] = 5
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
Cell In[3], line 1
----> 1 photon_array.x[:] = 5

File ~/mambaforge/envs/jax-galsim/lib/python3.13/site-packages/jax/_src/numpy/array_methods.py:622, in _unimplemented_setitem(self, i, x)
    618 def _unimplemented_setitem(self, i, x):
    619   msg = ("JAX arrays are immutable and do not support in-place item assignment."
    620          " Instead of x[idx] = y, use x = x.at[idx].set(y) or another .at[] method:"
    621          " https://docs.jax.dev/en/latest/_autosummary/jax.numpy.ndarray.at.html")
--> 622   raise TypeError(msg.format(type(self)))

TypeError: JAX arrays are immutable and do not support in-place item assignment. Instead of x[idx] = y, use x = x.at[idx].set(y) or another .at[] method: https://docs.jax.dev/en/latest/_autosummary/jax.numpy.ndarray.at.html
```

beckermr commented on tests/test_photon_array.py:
```diff
@@ -276,7 +297,10 @@ def test_photon_array():
     assert_raises(ValueError, pa2.copyFrom, pa1, slice(50,None), slice(50,None))
-    assert_raises(ValueError, pa2.copyFrom, pa1, 100, 0)
+    if is_jax_galsim():
+        pass
```
In JAX, no. Out of bounds indices generally do not raise. Here are the available behaviors: 

```
string specifying out-of-bound indexing mode. Options are:

"promise_in_bounds": (default) The user promises that indices are in bounds. No additional checking will be performed. In practice, this means that out-of-bounds indices in get() will be clipped, and out-of-bounds indices in set(), add(), etc. will be dropped.

"clip": clamp out of bounds indices into valid range.

"drop": ignore out-of-bound indices.

"fill": alias for "drop". For get(), the optional fill_value argument specifies the value that will be returned.
```

beckermr commented on tests/test_draw.py:
```diff
@@ -1097,19 +1113,31 @@ def test_shoot():
+        # jax galsim works not as well
+        np.testing.assert_array_almost_equal(image2.array, image1.array, decimal=10)
+    else:
+        np.testing.assert_array_almost_equal(image2.array, image1.array, decimal=12)
```
xref: https://github.com/GalSim-developers/GalSim/pull/1360

beckermr commented on tests/test_draw.py:
```diff
@@ -1097,19 +1113,31 @@ def test_shoot():
+        # jax galsim needs double images here
+        image1 = galsim.ImageD(32,32, init_value=100)
+    else:
+        image1 = galsim.ImageF(32,32, init_value=100)
```
xref: https://github.com/GalSim-developers/GalSim/pull/1360

beckermr commented on tests/test_noise.py:
```diff
@@ -311,17 +328,27 @@ def test_variable_gaussian_noise():
+        decimal = 1
     np.testing.assert_almost_equal(
-            var, big_vgn.var_image.array.mean(), 1,
+            var, big_vgn.var_image.array.mean(), decimal,
```
xref: https://github.com/GalSim-developers/GalSim/pull/1360

beckermr commented:
OK @rmjarvis I addressed all of the comments here. Thanks a bunch for the review! It definitely helped improve JAX-GalSim and helped to flush out at least one bug.
